.github/workflows/tests.yml

```diff
@@ -0,0 +1,74 @@
+name: Run all tests
+
+# Run this workflow every time a new commit is pushed to the repository
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+
+jobs:
+  run-all-tests:
+    name: Run all tests
+
+    # TODO: not sure what the best OS to run on is so ubuntu will do for now
+    runs-on: ubuntu-latest
+
+    strategy:
+      # Allows running the build multiple times with different configurations
+      matrix:
+        rust: ["stable", "beta"]
+        deny_warnings: ['', 1]
+
+        # TODO: I don't know how to make it run only the check_exercises.sh script
+        # For nightly rust, run the job with DENYWARNINGS=1
+        include:
+          - rust: nightly
+            deny_warnings: 1
+    
+    # Allow failures on nightly rust, or beta if DENYWARNINGS is enabled
+    continue-on-error: ${{ matrix.rust }} == "nightly" || ( ${{ matrix.rust }} == "beta" && DENYWARNINGS == 1 )
+
+    steps:
+      # Checks out a copy of your repository on the ubuntu-latest machine
+      - name: Checkout code
+        uses: actions/checkout@v2
+        env: 
+          DENYWARNINGS: ${{ matrix.deny_warnings }}
```

Comment:
It looks like you're going to need to copy the `env: DENYWARNINGS: ${{ matrix.deny_warnings }}` stuff into each appropriate job step, which should be the main check exercises and ensure stubs compile steps. The rest of the checks shouldn't care about this particular environment variable.

Reply:
Alternately, promoting the environment variable setting to the entire job is probably the better way to go.